### PR TITLE
[TAGrading: Feature] Instructor deletion of peer grading marks

### PR DESCRIPTION
**Pull request by jchulton**
### Please check if the PR fulfills these requirements:

* [ ] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)
* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
An instructor cannot edit peer's grading for an assignment
### What is the new behavior?
An instructor can remove all peer grading from specific graders through a peer grading edit modal as below: solves #5368

image

1. Click the edit icon (pencil) on the peer grading panel when grading as an instructor.
2. Select the student whose marks you want to delete in the dropdown menu
3. Click "Clear all grading"

### CODE REVIEW

**Comment by bmcutler** [comment]
@jchulton please resolve conflicts

**Comment by bmcutler** [comment]
I got this error when I tried to open the ta grading page for a student:


Oh no! Something irrecoverable has happened...

app\exceptions\OutputException (Code: 0) thrown in /usr/local/submitty/site/app/libraries/Output.php (Line 339) by:
            throw new OutputException("{$e->getMessage()} in {$e->getFile()}:{$e->getLine()}");


Message:
Unable to find template "grading/electronic/EditPeerComponentsForm.twig" (looked into: /usr/local/submitty/site/app/templates). in /usr/local/submitty/site/vendor/twig/twig/src/Loader/FilesystemLoader.php:250

Stack Trace:
#0 /usr/local/submitty/site/app/views/grading/ElectronicGraderView.php(1232): app\libraries\Output->renderTwigTemplate('grading/electronic/EditPeerComponentsForm.twig', Array)
#1 unknown file(unknown line): app\views\grading\ElectronicGraderView->renderPeerEditMarksPanel(app\models\gradeable\GradedGradeable)
#2 /usr/local/submitty/site/app/libraries/Output.php(201): call_user_func_array(Array, Array)
#3 /usr/local/submitty/site/app/views/grading/ElectronicGraderView.php(761): app\libraries\Output->renderTemplate('grading\ElectronicGrader', 'renderPeerEditMarksPanel', app\models\gradeable\GradedGradeable)
#4 unknown file(unknown line): app\views\grading\ElectronicGraderView->hwGradingPage(app\models\gradeable\Gradeable, app\models\gradeable\GradedGradeable, 1, 35, true, true, true, true, true, '1', -1, 'id', 'ASC', 'aphacker')
#5 /usr/local/submitty/site/app/libraries/Output.php(201): call_user_func_array(Array, Array)
#6 /usr/local/submitty/site/app/libraries/Output.php(170): app\libraries\Output->renderTemplate('grading\ElectronicGrader', 'hwGradingPage', app\models\gradeable\Gradeable, app\models\gradeable\GradedGradeable, 1, 35, true, true, true, true, true, 1, -1, 'id', 'ASC', 'aphacker')
#7 /usr/local/submitty/site/app/controllers/grading/ElectronicGraderController.php(1198): app\libraries\Output->renderOutput(Array, 'hwGradingPage', app\models\gradeable\Gradeable, app\models\gradeable\GradedGradeable, 1, 35, true, true, true, true, true, 1, -1, 'id', 'ASC', 'aphacker')
#8 unknown file(unknown line): app\controllers\grading\ElectronicGraderController->showGrading('peer_gradeable', 'aphacker', '', NULL, NULL, 'id', 'ASC', NULL, '-1')
#9 /usr/local/submitty/site/app/libraries/routers/WebRouter.php(217): call_user_func_array(Array, Array)
#10 /usr/local/submitty/site/app/libraries/routers/WebRouter.php(195): app\libraries\routers\WebRouter->run()
#11 /usr/local/submitty/site/public/index.php(127): app\libraries\routers\WebRouter::getWebResponse(Symfony\Component\HttpFoundation\Request, app\libraries\Core)

File: site/app/templates/grading/electronic/EditPeerComponentsForm.twig [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			    </div>
+	        {% endfor %}
+	    </div>
+    {% endfor %}
+    </br>
+
+<script>
+	function changeCurrentPeer(){
+		let peer = $('#edit-peer-select').val();
+		console.log('edit-peer-components-form-'+peer);
```
[the remark this answers is not in the document]
Comment (jchulton) [reply site/app/templates/grading/electronic/EditPeerComponentsForm.twig]: Removed

File: site/app/controllers/grading/ElectronicGraderController.php [hunk 2]
```diff
@@ -2477,4 +2477,28 @@ private function getStats(Gradeable $gradeable, User $grader, bool $full_stats,
             $total_total += $value * $num_components;
         }
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/gradeable/{gradeable_id}/grading/clear_peer_marks", methods={"POST"})
+     */
```
[the remark this answers is not in the document]
Comment (jchulton) [reply site/app/controllers/grading/ElectronicGraderController.php]: Added the above access control

**Comment by jchulton** [comment]
I've added the deletion of the overall comment and a warning message, as discussed

File: migration/migrator/data/course_tables.sql [hunk 3]
```diff
@@ -917,6 +917,7 @@ ALTER SEQUENCE public.student_favorites_id_seq OWNED BY public.student_favorites
 CREATE TABLE public.teams (
     team_id character varying(255) NOT NULL,
     user_id character varying(255) NOT NULL,
+    anon_id character varying(255),
```
[the remark this answers is not in the document]
Comment (jchulton) [reply migration/migrator/data/course_tables.sql]: Has been yanked out

File: site/app/models/Team.php [hunk 4]
```diff
@@ -83,7 +83,7 @@ public function __construct(Core $core, $details) {
      * @return string
      */
     public function getAnonId() {
-        if (empty($this->core->getQueries()->getTeamAnonId($this->getId()))) {
+        if (empty($this->core->getQueries()->getTeamAnonId($this->getId())) || $this->core->getQueries()->getTeamAnonId($this->getId())[$this->getId()] === null) {
```
[the remark this answers is not in the document]
Comment (jchulton) [reply site/app/models/Team.php:86]: Has also been yanked out

File: site/app/templates/grading/electronic/EditPeerComponentsForm.twig [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	        success: function(data) {
+	        	console.log("Successfully deleted peer marks");
+	        	window.location.reload(true);
+	        },
+	        error: function(e) {
+	            console.log("Failed to delete");
+	        }
+	    });
+	}
+</script>
```
[the remark this answers is not in the document]
Comment (jchulton) [reply site/app/templates/grading/electronic/EditPeerComponentsForm.twig]: moved to ta-grading.js

File: site/public/js/server.js [hunk 6]
```diff
@@ -1624,6 +1624,12 @@ function enableKeyToClick(){
   }
 }
 
+function newEditPeerComponentsForm() {
+    $('.popup-form').css('display', 'none');
+    let form = $("#edit-peer-components-form");
+    form.css("display", "block");
+    captureTabInModal("edit-peer-components-form");
+}
```
[the remark this answers is not in the document]
Comment (jchulton) [reply site/public/js/server.js]: Moved to ta-grading.js

**Comment by jchulton** [comment]
Overall comments should be deleted properly now

**Comment by jchulton** [comment]
Fixed the badges

**Pull request merged**